src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,386 @@
+    global_geometry::Geometry.AbstractGlobalGeometry = Geometry.ShallowSphericalGlobalGeometry(
+        radius,
+    ),
+    quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+    horizontal_layout_type = DataLayouts.IJFH,
+    h_mesh = Meshes.EquiangularCubedSphere(
+        Domains.SphereDomain{FT}(radius),
+        h_elem,
+    ),
+) where {FT}
+    @assert horizontal_layout_type <: DataLayouts.AbstractData
+    check_device_context(context, device)
+
+    z_boundary_names = (:bottom, :top)
+    h_topology = Topologies.Topology2D(context, h_mesh)
+    h_grid =
+        Grids.SpectralElementGrid2D(h_topology, quad; horizontal_layout_type)
+    z_domain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(z_min),
+        Geometry.ZPoint{FT}(z_max);
+        boundary_names = z_boundary_names,
+    )
+    z_mesh = Meshes.IntervalMesh(z_domain, stretch; nelems = z_elem)
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    vertical_grid = FiniteDifferenceGrid(z_topology)
+    return ExtrudedFiniteDifferenceGrid(
+        h_grid,
+        vertical_grid,
+        hypsography,
+        global_geometry,
+    )
+end
+
+"""
+    CubedSphereGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        radius::Real,
+        n_quad_points::Integer,
+        h_elem::Integer,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+        h_mesh = Meshes.EquiangularCubedSphere(Domains.SphereDomain(radius), h_elem),
+    )
+
+A convenience constructor, which builds a
+`SpectralElementGrid2D`.
+"""
+CubedSphereGrid(; kwargs...) = CubedSphereGrid(Float64; kwargs...)
+function CubedSphereGrid(
+    ::Type{FT};
+    radius::Real,
+    n_quad_points::Integer,
+    h_elem::Integer,
+    device::ClimaComms.AbstractDevice = ClimaComms.device(),
+    context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+    quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+    h_mesh = Meshes.EquiangularCubedSphere(
+        Domains.SphereDomain{FT}(radius),
+        h_elem,
+    ),
+    horizontal_layout_type = DataLayouts.IJFH,
+) where {FT}
+    @assert horizontal_layout_type <: DataLayouts.AbstractData
+    check_device_context(context, device)
+    h_topology = Topologies.Topology2D(context, h_mesh)
+    return Grids.SpectralElementGrid2D(h_topology, quad; horizontal_layout_type)
+end
+
+"""
+    ColumnGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        z_elem::Integer,
+        z_min::Real,
+        z_max::Real,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        stretch::Meshes.StretchingRule = Meshes.Uniform(),
+    )
+
+A convenience constructor, which builds a
+`FiniteDifferenceGrid` given.
```

Comment:
This sentence is incomplete

Reply:
I'm going to try and finish the interface before working on the docs